Marlin/src/module/planner.cpp

```diff
@@ -1143,7 +1143,7 @@ void Planner::recalculate_trapezoids() {
     block_t *prev = &block_buffer[prev_index];
 
     // If not dealing with a sync block, we are done. The last block is not a SYNC block
-    if (!(prev->flag & BLOCK_MASK_SYNC)) break;
+    if (!(prev->flag & BLOCK_MASK_SYNC) && !IS_PAGE(prev)) break;
```

Comment:
@colinrgodsey — Is this correct, something we missed in #17853 ?

Reply:
ooh yea, this should be added here. originally (iirc) i didn't update this because it was only getting called when buffering segments. but the page support doesnt actually prevent you from using traditional movement blocks, so this can still get called. good catch!

Reply:
Thanks for the reply. You never know if there is some subtle reason….